cli/azd/resources/alpha_features.yaml

```diff
@@ -0,0 +1,2 @@
+- id: terraform
+  description: "Provision Azure resources from terraform files."
```

Comment:
Do we need this as a separate configuration file? Or can this be described in code? I see that now have to maintain a weak-list of "keys", then a strong-list of constants in code. Can we avoid this duplication?

Reply:
Taking the benefit of `//go:embed` and a similar strategy as the one we already have for templates as part of the resources package.
If we want to use a different strategy, let's talk about it in the future and stay with the current existing strategy for now

Reply:
I'm supportive of this right now - I'm okay with the duplication and I think the source of truth of this being config rather than code (to match how we do our templates) will be useful down the line.

Reply:
@ellismg @vhvb1989  My worry is that we end up with simple authoring errors because of the duplication.

I write `terraforma` in `alpha_features.yml` and `terraform` in `features.go`

The runtime logic looks at `terraform`. Our documentation that we print out says `terraforma`. User follows the suggested command of `azd config set alpha.terraforma on`. This issue isn't easily caught. In fact, I would say we often skip past reviewing configuration files as we deem them less important than source files.

> Taking the benefit of //go:embed and a similar strategy as the one we already have for templates as part of the resources package.

For templates, we only have templates defined inside the JSON and nowhere else, which I believe is an important distinction. There isn't a runtime check for a template name inside source code `*.go` files.

Reply:
> I write terraforma in alpha_features.yml and terraform in features.go

This is at development time.  It must be discovered during testing.
Leaking this to a final version means the feature was not tested it at all?

Reply:
@vhvb1989 The UX is wrong in `list-alpha`, which is not easily caught. Your tests would all pass in theory.